### DISCUSSION TITLE
How to combine with VueDraggable?

### DISCUSSION THREAD

**Discussion by derwaldgeist** [simonwep/viselect#229] [Q&A]
I'd like to combine this library with `VueDraggable`, so the items are both selectable and draggable. Using the Vue version does not work, since it sets up its own `div` around the items (which prevents VueDraggble from being able to drag individual items).

So I tried to use the plain vanilla version instead. But I could not get it working. I tried to setup the `SelectionArea` in an `onMounted()` hook, but that did not have any effect.

I had tried to use the same container as I use for VueDraggable, but even if I do not configure a special container class, no container is being constructed.

(It is also important that the items can be added or removed dynamically.)

**Comment by simonwep** [comment]
> So I tried to use the plain vanilla version instead. But I could not get it working. I tried to setup the SelectionArea in an onMounted() hook, but that did not have any effect.

It's best to watch the `ref` to the element you want the selection being applied and destroy/re-create the selection instance on every change. Otherwise this might break reactivity if the `ref` to the element changes.

> I had tried to use the same container as I use for VueDraggable, but even if I do not configure a special container class, no container is being constructed.

Hm, could you share some code? Viselect doesn't create any container on it's own, see the [vanilla documentation](https://github.com/simonwep/selection/tree/master/packages/vanilla#getting-started) of how to use it.
